synapse/handlers/saml_handler.py

```diff
@@ -80,20 +80,30 @@ def __init__(self, hs: "HomeServer"):
 
         self._sso_handler = hs.get_sso_handler()
 
-    def handle_redirect_request(
-        self, client_redirect_url: bytes, ui_auth_session_id: Optional[str] = None
-    ) -> bytes:
+    async def handle_redirect_request(
+        self,
+        request: SynapseRequest,
+        client_redirect_url: Optional[bytes],
+        ui_auth_session_id: Optional[str] = None,
+    ) -> str:
```

Comment:
it always returned a `str`, but the declaration was incorrect. Again, it worked because you can `request.redirect` to a bytes or a str.